src/plugins/es_ui_shared/public/request/use_request.helpers.tsx

```diff
@@ -86,40 +86,50 @@ export const createUseRequestHelpers = (): UseRequestHelpers => {
   const hookResult = {} as UseRequestResponse;
   const sendRequestSpy = sinon.stub();
 
-  const setupUseRequest = (config: SendRequestConfig) => {
-    const httpClient = {
-      post: (path: string, options: HttpFetchOptions) => {
-        return new Promise((resolve, reject) => {
-          // Increase the time it takes to resolve a request so we have time to inspect the hook
-          // as it goes through various states.
-          setTimeout(() => {
-            try {
-              resolve(sendRequestSpy(path, options));
-            } catch (e) {
-              reject(e);
-            }
-          }, REQUEST_TIME);
-        });
-      },
-    };
-
-    const TestComponent = ({ requestConfig }: { requestConfig: SendRequestConfig }) => {
-      const { isInitialRequest, isLoading, error, data, sendRequest } = useRequest(
-        httpClient as HttpSetup,
-        requestConfig
-      );
+  const httpClient = {
+    post: (path: string, options: HttpFetchOptions) => {
+      return new Promise((resolve, reject) => {
+        // Increase the time it takes to resolve a request so we have time to inspect the hook
+        // as it goes through various states.
+        setTimeout(() => {
+          try {
+            resolve(sendRequestSpy(path, options));
+          } catch (e) {
+            reject(e);
+          }
+        }, REQUEST_TIME);
+      });
+    },
+  };
 
-      hookResult.isInitialRequest = isInitialRequest;
-      hookResult.isLoading = isLoading;
-      hookResult.error = error;
-      hookResult.data = data;
-      hookResult.sendRequest = sendRequest;
+  const setupUseRequest = (
+    config: SendRequestConfig,
+    onChange: (hookData: UseRequestResponse) => void = (hookData) => {
+      hookResult.isInitialRequest = hookData.isInitialRequest;
```

Comment:
I copied over your logic here in a "default" `onChange` handler to not make a bigger refactor.